### DISCUSSION TITLE
[Proposal] Resolve relative assets on Markdown files

### DISCUSSION THREAD

**Discussion by kasvith** [withastro/roadmap#334] [Proposal]
### Problem:

When using Astro for a content site(blog) Markdown is often used. Currently, Astro cannot resolve files outside the `public` or `assets` directories.

I don't want to keep my content inside the `src` folder, **app** and **content** are two separate things.

Here is how you can usually organize content clean and separate from the app source code.

```
src
content
 ┗ blog
 ┃ ┗ first-post
 ┃ ┃ ┣ imgs
 ┃ ┃ ┃ ┗ cover.jpg
 ┃ ┃ ┗ post.md
```

```md
---
title: Hello, World
---

# Hi there!

This is your first markdown page. It probably isn't styled much, although
Markdown does support **bold** and _italics._

To learn more about adding a layout to your page, read the next section on **Markdown Layouts.**

![Img](imgs/cover.jpg)
```

### Solution

When you find a relative resource file referenced in `.md` or `.mdx`, Astro should automatically detect this and serve the resource to the client without additional steps. This is the most convenient way to handle it. (maybe introduce a new config param to disable this behaviour).

On production build, Astro should treat these resources, as usual

**Note:** Hugo, Gridsome, Gatsby supports this out of the box

**Comment by Temez1** [comment]
I agree & this would be good for integrations as well to be able to separate the content away from source code. Eg. In my case I'm planning to integrate Stackbit later to the project that mentions

> Stackbit's only real requirement is that content must be structured and separated from code.

It would be great to understand what limits Astro to do this currently & how this relates to their Image optimization etc. I'm interested to optimize the images in md files as well, which currently is out of the box possible only in [.mdx files](https://docs.astro.build/en/guides/images/#using-in-mdx)

**Reply by kasvith** [reply]
Even for MDX this feels like a hack to me. Naturally Astro should resolve this.
Anyway for MD we made an integration to save you from this https://github.com/userquin/astro-md-image-plugin

IDK why astro doesn't support this by default

**Reply by Temez1** [reply]
I suspect this will be supported in the future when the project matures.

Thanks for the integration. However, I'm going with the Astro official Image integration as it handles optimization for images and works out of the box.

I will just put my content folder in the src/ folder and use mdx files instead of md.

**Reply by kasvith** [reply]
Optimizing images with Astro is something different from what we achieved here. Tbh in Gridsome this was fully automatic(content, images in markdown etc) all were optimized(its even better than Astro images, gridsome also blurs images)

Lets see if we can achieve this also, its like out of scope of the integration

**Reply by Temez1** [reply]
Yeah, responsive optimized images is my primary concern. Having the content folder outside src is secondary as I believe eg. Stackbit doesn't mind where the content dir lives.

And I believe astro Image will get better over time. No idea how it compares to next image currently.

**Comment by florian-lefebvre** [comment]
It's now possible https://docs.astro.build/en/guides/images/#images-in-markdown-files
